### ISSUE TITLE
Creating a cluster with invalid snapshot/retention time and editing to right values does not start snapshot saving

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#16553]
Environment information: v2.1.2-rc9

Steps to reproduce (least amount of steps as possible):

1. Create a single node DO cluster(RKE) with snapshots enabled every 5m and invalid value 1d for retention period
etcd-rolling-snapshots starts and exits as below: No snapshots are created
```
6b19995bbf07        rancher/rke-tools:v0.1.15                                                                                                 "/opt/rke-tools/rk..."   8 minutes ago       Exited (0) 8 minutes ago                       etcd-rolling-snapshots
befe20c5dff6        rancher/coreos-etcd:v3.2.18                                                                                               "/usr/local/bin/et..."   8 minutes ago       Up 8 minutes                                   etcd

```
Rancher logs: 
```
2018/11/12 21:17:33 [INFO] cluster [c-h4bdc] provisioning: [etcd] Successfully pulled image [rancher/coreos-etcd:v3.2.18] on host [68.183.50.22]
2018/11/12 21:17:36 [INFO] cluster [c-h4bdc] provisioning: [etcd] Successfully started [etcd] container on host [68.183.50.22]
2018/11/12 21:17:36 [INFO] cluster [c-h4bdc] provisioning: [etcd] Saving snapshot [etcd-rolling-snapshots] on host [68.183.50.22]
2018/11/12 21:17:36 [INFO] cluster [c-h4bdc] provisioning: [etcd] Successfully started [etcd-rolling-snapshots] container on host [68.183.50.22]
2018/11/12 21:17:37 [INFO] cluster [c-h4bdc] provisioning: [certificates] Successfully started [rke-bundle-cert] container on host [68.183.50.22]
2018/11/12 21:17:37 [INFO] cluster [c-h4bdc] provisioning: [certificates] successfully saved certificate bundle [/opt/rke/etcd-snapshots//pki.bundle.tar.gz] on host [68.183.50.22]
2018/11/12 21:17:37 [INFO] cluster [c-h4bdc] provisioning: [etcd] Successfully started [rke-log-linker] container on host [68.183.50.22]
2018/11/12 21:17:38 [INFO] cluster [c-h4bdc] provisioning: [remove/rke-log-linker] Successfully removed container on host [68.183.50.22]
2018/11/12 21:17:38 [INFO] cluster [c-h4bdc] provisioning: [etcd] Successfully started etcd plane..

```
3. Edit the cluster and fix the values to 5m and 30m for snapshot creation and retention respectively
 etcd-rolling-snapshots fails to start. Snapshots are not saved in /etcd/rke/etcd-snapshots directory

```
root@soudotestcluster1:/opt/rke/etcd-snapshots# sudo docker ps -a|grep etcd
6b19995bbf07        rancher/rke-tools:v0.1.15                                                                                                 "/opt/rke-tools/rk..."   32 minutes ago      Exited (0) 7 minutes ago                        etcd-rolling-snapshots
befe20c5dff6        rancher/coreos-etcd:v3.2.18                                                                                               "/usr/local/bin/et..."   33 minutes ago      Up 32 minutes                                   etcd



```

**Comment by alena1108** [comment]
Fix is available in rancher/rancher:2.1.2-rc11

**Comment by soumyalj** [comment]
Tested with rancher v2.1.2-rc11 
Using API created a single node DO cluster(RKE) with snapshots enabled every 5m and invalid value 1d for retention period. No snapshots were created.
Edited the cluster and provided the right values 5m/30m for creation/retention time. etcd-rolling-snapshots was created successfully and snapshots were created every 5m as expected.

```
b32a3145b9c2        rancher/rke-tools:v0.1.15                                                                                                 "/opt/rke-tools/rk..."   11 minutes ago      Up 11 minutes                                     etcd-rolling-snapshots
51e7bf974c03        rancher/coreos-etcd:v3.2.18                                                                                               "/usr/local/bin/et..."   35 minutes ago      Up 35 minutes                                     etcd

```
Snapshots below:
```
root@test-98056:/opt/rke/etcd-snapshots# ls
2018-11-14T20:47:43Z_etcd  2018-11-14T20:52:43Z_etcd  pki.bundle.tar.gz
root@test-98056:/opt/rke/etcd-snapshots# 


```
